packages/contracts/test/scripts/MigrateMetapool.t.sol

```diff
@@ -0,0 +1,249 @@
+            so we are unable to migrate all funds.
+        */
+
+        assertEq(v2Curve3PreBalance, (v2Curve3PostBalance + v3Curve3Balance));
+
+        /// Same as above but for Ubiquity Dollar Token
+        assertEq(
+            v2DollarPreBalance,
+            v2DollarPostBalance +
+                v3DollarBalance +
+                dollarToken.balanceOf(admin) -
+                adminDollarPreBalance
+        );
+    }
+
+    function testV2WithdrawVsV3() public {
+        address user = 0x4007CE2083c7F3E18097aeB3A39bb8eC149a341d;
+        uint256[] memory tokens = bonds.holderTokens(user);
+
+        uint256 dollarTokenPreBalance = dollarToken.balanceOf(user);
+        BondingShareV2.Bond memory bond = bonds.getBond(tokens[0]);
+        uint256 withdraw = bond.lpAmount;
+
+        vm.roll(block.number + 10483200);
+
+        vm.startPrank(user);
+        staking.removeLiquidity(withdraw, tokens[0]);
+        uint256 v2Balance = v2Metapool.balanceOf(user);
+        v2Metapool.remove_liquidity(v2Balance, [uint256(0), uint256(0)]);
+        vm.stopPrank();
+
+        uint256 userDollarV2 = dollarToken.balanceOf(user) -
+            dollarTokenPreBalance;
+        uint256 userLPV2 = curve3Token.balanceOf(user);
+
+        vm.revertTo(snapshot);
+
+        _migrate();
+
+        vm.roll(block.number + 10483200);
+        BondingShareV2.Bond memory bondV3 = bonds.getBond(tokens[0]);
+        uint256 withdrawV3 = bondV3.lpAmount;
+        vm.startPrank(user);
+        staking.removeLiquidity(withdrawV3, tokens[0]);
+        uint256 v3Balance = v3Metapool.balanceOf(user);
+
+        v3Metapool.remove_liquidity(v3Balance, [uint256(0), uint256(0)]);
+        vm.stopPrank();
+
+        uint256 userDollarV3 = dollarToken.balanceOf(user) -
+            dollarTokenPreBalance;
+        uint256 userLPV3 = curve3Token.balanceOf(user);
+
+        uint256 v3DollarBalance = dollarToken.balanceOf(address(v3Metapool));
+        uint256 v3Curve3Balance = curve3Token.balanceOf(address(v3Metapool));
+
+        console.log("Metapool LP Tokens withdrawn pre migration: ", v2Balance);
+        console.log("Metapool LP Tokens withdrawn post migration: ", v3Balance);
+        console.log(
+            "UbiquityDollar Tokens withdrawn pre migration: ",
+            userDollarV2
+        );
+        console.log("Curve3 Tokens withdrawn pre migration: ", userLPV2);
+        console.log(
+            "UbiquityDollar Tokens withdrawn post migration: ",
+            userDollarV3
+        );
+        console.log("Curve3 Tokens withdrawn post migration: ", userLPV3);
+        console.log(
+            "Curve Metapool V3 UbiquityDollar balance after withdrawal: ",
+            v3DollarBalance
+        );
+        console.log(
+            "Curve Metapool V3 Curve3 balance after withdrawal: ",
+            v3Curve3Balance
+        );
+
+        assertGe(v3Balance, v2Balance);
+        assertLt(userDollarV3, userDollarV2);
```

Comment:
there is a difference of about 3907 dollar for the user between when withdrawing with v2 and v3
I guess that is because when migrating to v3 as v2 pool is unbalanced we don't have the same dollar amount in the v3 pool
but we have to make sure that  the user is not loosing one penny so after migrating user should withdraw from V2 AND v3 and we should het the same amount as if there were no migration and user withdraw all from v2
In short we should have strong assertion in this test "Greater than" and "Less than" are not enough to really make sure nothing is lost for the users 
Also it is worth checking that all dollar in pools stay the same e.g:
- in V2 no migration  
V2 pool's dollar balance =  X dollars
- migration to v3
V2 pool's dollar balance =  Y dollars
V3 pool's dollar balance =  Z dollars
X = Y +Z

Reply:
![image](https://user-images.githubusercontent.com/16853695/221511878-923b2bf1-004f-454e-beb4-4af1fbbc42a3.png)

Reply:
Only way I can think of to resolve that is do a single sided liquidity add after launching the TWAPOracle for v3 pool

Reply:
if we withdraw everything from V2 and V3 (so after migration) we should get the same amount of dollar than if we withdraw everything from v2 (without migration) right ?

Reply:
Ok lp tokens amount make less sense than the coins behind. Can you check the exact amount of dollar/3crv behind?

Reply:
I added that in the logs earlier as well. 
```Amount of UbiquityDollar Tokens in Curve Metapool V2 after migration:  169833975952106434827
  Amount of Curve3 Tokens in Curve Metapool V2 after migration:  142863822020848582307```

Reply:
The X=Y+Z checks out. `assertEq()` for those is at lines 121 and 124

Reply:
In the withdrawal comparison between pre and post migration, there's only a difference of ~.20 in `uAD` but `curve3` is ~143. This is fairly significant and I don't think it can be remedied without adding additional `curve3` to the pool.

Reply:
I was looking at the wrong numbers. Difference in withdrawal is ~.2 less for each token after migration. Comes out to to 0.002% less.